repo.go

```diff
@@ -526,6 +534,60 @@ func (r *Repo) Sign(name string) error {
 	return r.local.SetMeta(name, b)
 }
 
+// AppendSignature allows users to append a signature generated with an external tool.
+// The name must be a valid manifest name, like root.json.
+func (r *Repo) AppendSignature(name string, signature data.Signature) error {
+	role := strings.TrimSuffix(name, ".json")
+	if !verify.ValidRole(role) {
+		return ErrInvalidRole{role}
+	}
+
+	// Check key ID is in valid for the role.
+	db, err := r.db()
+	if err != nil {
+		return err
+	}
+	roleData := db.GetRole(role)
+	if roleData == nil {
+		return ErrInvalidRole{role}
+	}
+	if !roleData.ValidKey(signature.KeyID) {
+		return verify.ErrInvalidKey
+	}
+
+	s, err := r.SignedMeta(name)
+	if err != nil {
+		return err
+	}
+
+	// Add signature if one doesn't already exist with the same key ID.
+	found := false
+	for _, sig := range s.Signatures {
+		if sig.KeyID == signature.KeyID {
+			found = true
+		}
+	}
+	if !found {
```

Comment:
Should we return an error if found? Otherwise, function can "fail" silently, surprising the user.

Reply:
of, this is adding the signature if it's not already there. so this is expected actually...

Reply:
No, no, I understand. I guess, what should we make of the user who appends a new signature w/o checking that there is an old one (which is perhaps obsolete now)? We could punt it to them to check, which is not a wrong answer.

Reply:
Hmmm whenever there is a repo change, setMeta will remove existing signatures and re-sign https://github.com/theupdateframework/go-tuf/blob/22a94818d17b266491bd7bedb8f311c7ebced9c5/repo.go#L482

In the case where you aren't storing keys and you are using external tools, you'll have no signatures when you setMeta. So I'm not sure this is possible? It should not. I will also add a test for this case.

(This makes me think we should be overriding though, just in case, for a latest signature)

Reply:
We just have to make a decision: either error on duplicate signatures, or do not allow duplicate signatures. If the latter, I recommend re-adding the logic, and renaming the function to follow principle of least surprise. I will let you decide 🙂

Reply:
Agreed, I think it's better for AddOrUpdate now.